src/state/validation.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for functions containing charm utilities."""
+
+import functools
+import logging
+import typing
+
+import ops
+
+from state.config import InvalidCharmConfigError
+
+logger = logging.getLogger(__name__)
+
+C = typing.TypeVar("C", bound=ops.CharmBase)
+
+
+def validate_config_and_integration(
+    defer: bool = False,
```

Comment:
Isn't this arg always false? Why keeping it?

Reply:
I'm keeping it here because it'll be used for tls-certificates related hooks in future PRs and in those we'd want to have `defer=True` (mainly for `certificates-relation-*` hook)

Reply:
Also this was copied over from the gateway-api-integrator charm so I was thinking to keep it as is since it'll be used relatively soon.